### PR TITLE
N°3455 - Passing json_data as file to REST API

### PR DESCRIPTION
**Pull request by odain-cbd**
It is always error prone to manipulate JSON content as string especially when calling iTop API under CLI: 
- JSON format can be invalid. 
- Unix shell can interpret JSON inner content and truncate your query. 
- You can have encoding issues or wonder whether caracters have to be html-ized...

With this pull request we can save a JSON file and pass its content as file to any http client (curl, wget, PHP curl_exec, ...) as json_data parameter to call iTop REST API.  

Example:
curl -X POST  -F 'version=1.3' -F 'auth_user=XXX' -F 'auth_pwd=YYY'  http://localhost/iTop/webservices/rest.php -F 'json_data=**@**/tmp/request.json'

where  /tmp/request.json contains a valid json request (cf https://wiki.combodo.com/doku.php?id=latest:advancedtopics:rest_json&s[]=rest) 

this enhancement is described in bug N°3455 (Combodo internal).

### CODE REVIEW

File: webservices/rest.php [hunk 1]
```diff
@@ -70,7 +70,21 @@ function json_last_error_msg() {
 
 $sVersion = utils::ReadParam('version', null, false, 'raw_data');
 $sOperation = utils::ReadParam('operation', null);
-$sJsonString = utils::ReadParam('json_data', null, false, 'raw_data');
+if(isset($_FILES['json_data']['name']) && !empty($_FILES['json_data']['name']))
```
Comment (bruno-ds) [inline webservices/rest.php]: maybe check directly `tmp_name` ?

**Review by bruno-ds — DISMISSED** [review]

**Comment by Hipska** [comment]
Why is it closed? Not relevant/wanted anymore?

**Comment by odain-cbd** [comment]
that enhancement has been reviewed and accepted by the team.

indeed this code has been duplicated and merged into develop from branch feature/restapi-pass-json-asfile2.

why? I needed to reword the commit messages easily...

**Comment by Hipska** [comment]
🙄

**Pull request closed without merging**